### PR TITLE
DP-16475 Sanitize user roles and name

### PR DESCRIPTION
**Pull request by mrossi113**
**Description:**
Removing user names and roles from the database.

**Jira:**
https://jira.mass.gov/browse/DP-16475

**To Test:**
- [ ] Add steps to test this feature

**Screenshots/GIFs:**
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

---

[Peer Review Checklist](https://github.com/massgov/opemass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

**Review by weitzman — CHANGES_REQUESTED** [review]
Instead of creating new Drush commands, I think it makes sense to extend sql:sanitize, like the rest of that file does. See how the new --sanitize-entities and --sanitize-keyvalue options are implemented.

File: drush/Commands/SanitizationCommands.php [hunk 1]
```diff
@@ -195,4 +196,42 @@ private function getEntityStorage(string $type) {
...
+   * Sanitize the database table for the username.
+   *
+   * @hook post-command sql-sanitize
+   */
+  public function userName($result, CommandData $commandData) {
+    if (!$commandData->input()->getOption('sanitize-name')) {
+      return;
+    }
+    // User data table updated.
+    $name = Database::getConnection()->truncate('users_data table')->execute();
```
Comment (weitzman) [inline drush/Commands/SanitizationCommands.php]: User names are stored in `users_field_data`, not users_data. Also, we dont want to truncate that whole table, or else all our content will have no owner. I think we want a method very similar to https://github.com/drush-ops/drush/blob/40092d5dcf6f850fdf15d817c33602fe92c67c7c/src/Drupal/Commands/sql/SanitizeUserTableCommands.php#L26-L84. That sanitizes email addresses and passwords.

**Review by weitzman — CHANGES_REQUESTED** [review]

File: drush/Commands/SanitizationCommands.php (hunk 1 above)
Comment (mrossi113) [reply drush/Commands/SanitizationCommands.php]: I've updated the username sanitization option.

**Review by weitzman — COMMENTED** [review]
userName looks good. You now need to declare the `sanitize-name` option by adding it to options() and messages() methods in same file

**Comment by mrossi113** [comment]
Added the `sanitize-name` option to the `options()` and `messages()` methods.

**Pull request merged**